Lesson_16/l16_h01_class.py

```diff
@@ -0,0 +1,153 @@
+"""
+HW:16[Volkov Anton].
+
+# write class for bank card.
+# Class should reflect card lifecycle, card operations etc.
+# use CVV, PIN, Name, Surname , end date, card_num as initial params
+# class should have in addition to common logic some class attributes,
+# as minimum one classmethod and
+# as minimum  one staticmethod, two or more getters/setters
+# do not forget about block if __name__ == '__main__':
+"""
+
+
+class MonoBankCard:
+    """Class for creating cards."""
+
+    PIN_MAX_SIZE = 4
+    BANK_ID = 5375
+
+    def __init__(self, card_num, cvv, pin, name, surname, end_date, balance=0):
+        """Initialize class."""
+        self._card_num = card_num
+        self.__cvv = cvv
+        self.__pin = pin
+        self._name = name
+        self._surname = surname
+        self._end_date = end_date
+        self._lvl_card = 'Universal'
+        self._balance = balance
+        self._is_blocked = False
+
+    @property
+    def card_num(self):
+        """Get the card number."""
+        return self._card_num
+
+    @card_num.setter
+    def card_num(self, new_card_num):
+        """Set the card number."""
+        if self.validate_card_number(new_card_num):
+            self._card_num = new_card_num
+        else:
+            print('Invalid card number.')
+
+    @property
+    def cvv(self):
+        """Get the CVV."""
+        return self.__cvv
+
+    @property
+    def pin(self):
+        """Get the PIN."""
+        return self.__pin
+
+    @pin.setter
+    def pin(self, new_pin):
+        """Set the PIN."""
+        if new_pin.isdigit() and len(new_pin) == MonoBankCard.PIN_MAX_SIZE:
+            self.__pin = new_pin
+        else:
+            print(f'PIN should be {MonoBankCard.PIN_MAX_SIZE} size.')
+
+    def block_card(self):
+        """Block the card."""
+        self._is_blocked = True
+        print('Card is blocked.')
+
+    def unblock_card(self):
+        """Unblock the card."""
+        self._is_blocked = False
+        print('Card is unblocked.')
+
+    def pay(self, amount):
+        """Make a payment."""
+        if not self._is_blocked and self._balance >= amount:
```

Comment:
Not critical ut for future each check better to do within separate logic block.. For example check expiration, check amount etc. In future all these blocks could be refactored to separate functions if needed